### PR TITLE
Add an example Mapper

### PR DESCRIPTION
**Pull request by AlCutter**
This mapper reads a CT Log, and writes a map of DomainName to indicies
of log entries where that domain name was present in certificates or
precerts.

### CODE REVIEW

File: examples/ct/ct_mapper/ct_mapper.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+var sourceLog = flag.String("source", "https://ct.googleapis.com/testtube", "Source CT Log")
+var mapServer = flag.String("map_server", "", "host:port for the map server")
+var mapID = flag.Int("map_id", -1, "Map ID to write to")
+var batchSize = flag.Int("batch_size", 256, "Max number of entries to process at a time from the CT Log")
+
+//TODO(al): factor this out into a reuable thing.
```
Comment (Martin2112) [inline examples/ct/ct_mapper/ct_mapper.go]: typo

File: examples/ct/ct_mapper/ct_mapper.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+	"github.com/google/certificate-transparency/go/x509"
+	"github.com/google/trillian"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+var sourceLog = flag.String("source", "https://ct.googleapis.com/testtube", "Source CT Log")
+var mapServer = flag.String("map_server", "", "host:port for the map server")
+var mapID = flag.Int("map_id", -1, "Map ID to write to")
+var batchSize = flag.Int("batch_size", 256, "Max number of entries to process at a time from the CT Log")
```
Comment (Martin2112) [inline examples/ct/ct_mapper/ct_mapper.go]: log_batch_size maybe?

File: examples/ct/ct_mapper/ct_mapper.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+	if meta == nil {
+		meta = &trillian.MapperMetadata{}
+	}
+	startEntry = meta.HighestFullyCompletedSeq + 1
+	endEntry := startEntry + int64(*batchSize)
+
+	glog.Infof("Fetching entries [%d, %d] from log", startEntry, endEntry)
+
+	// Get the entries from the log:
+	logEntries, err := m.ct.GetEntries(startEntry, endEntry)
```
Comment (Martin2112) [inline examples/ct/ct_mapper/ct_mapper.go]: this can return a truncated range? does it matter?

File: examples/ct/ct_mapper/ct_mapper.go (hunk 1 above)
Comment (AlCutter) [inline examples/ct/ct_mapper/ct_mapper.go]: done.

File: examples/ct/ct_mapper/ct_mapper.go (hunk 2 above)
Comment (AlCutter) [inline examples/ct/ct_mapper/ct_mapper.go]: Done

File: examples/ct/ct_mapper/ct_mapper.go (hunk 3 above)
Comment (AlCutter) [inline examples/ct/ct_mapper/ct_mapper.go]: Nope, should be fine

**Pull request merged**